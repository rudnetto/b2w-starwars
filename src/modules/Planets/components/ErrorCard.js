// @flow
import React from 'react';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import Components from './Styled';

const ErrorCard = (props: Props) => {
  return (
    <Components.Card raised>
      <CardHeader title="Error" />
      <CardContent>
        <Typography color="inherit" variant="subheading">
          There was an unexpected error. Please, try again.
        </Typography>
      </CardContent>
    </Components.Card>
  );
}

export default ErrorCard;
