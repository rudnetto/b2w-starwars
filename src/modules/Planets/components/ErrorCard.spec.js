import React from 'react';
import { shallow } from 'enzyme';
import ErrorCard from './ErrorCard';

describe('ErrorCard test suite', () => {
  it('should render with default props', () => {
    const element = shallow(<ErrorCard />);
    expect(element).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const element = shallow(<ErrorCard title="Error" message="This is an error" />);
    expect(element).toMatchSnapshot();
  });
});
