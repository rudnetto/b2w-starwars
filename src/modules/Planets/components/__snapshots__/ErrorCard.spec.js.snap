// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard test suite should render with custom props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorCard
    message="This is an error"
    title="Error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardHeader)
          title="Error"
/>,
        <WithStyles(CardContent)>
          <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
          >
                    This is an error
          </WithStyles(Typography)>
</WithStyles(CardContent)>,
      ],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            color="inherit"
            variant="subheading"
>
            This is an error
</WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is an error",
            "color": "inherit",
            "variant": "subheading",
          },
          "ref": null,
          "rendered": "This is an error",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardHeader)
            title="Error"
/>,
          <WithStyles(CardContent)>
            <WithStyles(Typography)
                        color="inherit"
                        variant="subheading"
            >
                        This is an error
            </WithStyles(Typography)>
</WithStyles(CardContent)>,
        ],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              color="inherit"
              variant="subheading"
>
              This is an error
</WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "This is an error",
              "color": "inherit",
              "variant": "subheading",
            },
            "ref": null,
            "rendered": "This is an error",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ErrorCard test suite should render with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorCard
    message="There was an unexpected error. Please, try again."
    title="Error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardHeader)
          title="Error"
/>,
        <WithStyles(CardContent)>
          <WithStyles(Typography)
                    color="inherit"
                    variant="subheading"
          >
                    There was an unexpected error. Please, try again.
          </WithStyles(Typography)>
</WithStyles(CardContent)>,
      ],
      "raised": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": "Error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            color="inherit"
            variant="subheading"
>
            There was an unexpected error. Please, try again.
</WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "There was an unexpected error. Please, try again.",
            "color": "inherit",
            "variant": "subheading",
          },
          "ref": null,
          "rendered": "There was an unexpected error. Please, try again.",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardHeader)
            title="Error"
/>,
          <WithStyles(CardContent)>
            <WithStyles(Typography)
                        color="inherit"
                        variant="subheading"
            >
                        There was an unexpected error. Please, try again.
            </WithStyles(Typography)>
</WithStyles(CardContent)>,
        ],
        "raised": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              color="inherit"
              variant="subheading"
>
              There was an unexpected error. Please, try again.
</WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "There was an unexpected error. Please, try again.",
              "color": "inherit",
              "variant": "subheading",
            },
            "ref": null,
            "rendered": "There was an unexpected error. Please, try again.",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
